### PR TITLE
For #3722: Invoke search on long press home button

### LINKED ISSUE #3722: Open searchbox when trigerring assistant
### Why/User Benefit/User Problem

To search the web and avoid Google, i'm currently using the classical Firefox for Android and use it with the Assistant Button (which place the cursor on the searchbox and opens the keyboard), and make the app ready to search. This is very convenient.

Currently thanks to #390 Fenix can be mapped to the assistant button, but doesn't open the keyboard. It's __definitely__ polishing and not a top priority, but would be great if it's the same behavior as the old Firefox.

### Acceptance Criteria (how do I know when I’m done?)

Place the text cursor on the searchbox and open the keyboard when I open Fenix with the assistant button.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-12335)

### PR DESCRIPTION
**Pull request by boek**
When opening Fenix as an assistance, automatically take them to search.

Co-authored-by: s-ankur <s.ankursonawane@gmail.com>

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #3722

### CODE REVIEW

**Comment by gabrielluong** [comment]
@boek I could've swore that you asked product if this was something they wanted, but alas I couldn't find any traces of it anywhere from GitHub to Jira nor Slack. I just wanted to double check that Product is aware and okay with landing the feature.

**Comment by boek** [comment]
@gabrielluong good catch! It was in Slack https://mozilla.slack.com/archives/CG2FGG1ST/p1670029593236359. I will try to direct everything into Github/Bugzlla going forward!

File: app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt:31]: Drive-by thought - Since this revolves around search, I wanted to make sure we're mindful about collecting telemetry as well. I do remember doing something like this to ensure we have the right telemetry collected. Without looking too deeply into how the metrics are collected, I am wondering if you had a chance to test the telemetry collected here and if this was the only change we needed for telemetry?

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt:31]: We got a wiki for testing telemetry at https://github.com/mozilla-mobile/fenix/wiki/Test-telemetry-pings

**Comment by gabrielluong** [comment]
I am thinking we should consider waiting until after the merge date to land this, how does that sound with you?

File: app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt:31]: Since this is a new feature - we should file a separate issue for collecting telemetry instead of feature-creeping this one.

That said, it should be a fast-followup to your points as well about search.

**Comment by boek** [comment]
@jonalmeida @gabrielluong good catch! I’ll make sure that we’re testing telemetry and get an appropriate data review up. I’ll go ahead and throw the do not land label on for now until the next release gets cut as well

File: app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt]: I tested the telemetry and it worked, but I agree let's follow up telemetry with a separate issue: https://bugzilla.mozilla.org/show_bug.cgi?id=1808043

File: app/src/test/java/org/mozilla/fenix/home/intent/AssistIntentProcessorTest.kt
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/home/intent/AssistIntentProcessorTest.kt:28]: We moved towards using [GIVEN-WHEN-THEN style](https://martinfowler.com/bliki/GivenWhenThen.html) for our test descriptions. Could we update these?

**Review by gabrielluong — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/home/intent/AssistIntentProcessorTest.kt
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/home/intent/AssistIntentProcessorTest.kt:53]:
```suggestion
                    searchAccessPoint = MetricsUtils.Source.None,
```

**Pull request merged**
